### PR TITLE
JU-499 add @Validated annotation to NewSubmissionController

### PR DESCRIPTION
**Pull request by mnycz-ch**
https://companieshouse.atlassian.net/browse/JU-499 
move PresenterApi back to efs-submission-api

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by akunati-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by PhilipGlover — COMMENTED** [review]
I'm not understanding the reason for swopping the model in the private-api-sdk (https://github.com/companieshouse/private-api-sdk-java/blob/main/src/main/java/uk/gov/companieshouse/api/model/efs/submissions/PresenterApi.java) and placing it in here.

**Comment by mnycz-ch** [comment]
HI @PhilipGlover, it is working fine in my local as well. 
But for some reason we do not understand it is failing 4 karate test no reaching this point and returning 200 instead of 400.
You can test it as well in postman:
POST https://internalapi.cidev.aws.chdev.org/efs-submission-api/submissions/new 
Body {}
will return 200.

We want to check if it is some issue after upgrade for reaching that in private-api-sdk and move it back to efs-submission-api to run karate test and postman.

**Review by clewis1 — APPROVED** [review]
Looks good, all tests pass.

File: src/main/java/uk/gov/companieshouse/efs/api/submissions/controller/NewSubmissionController.java [hunk 1]
```diff
@@ -13,9 +13,9 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import uk.gov.companieshouse.api.model.efs.submissions.PresenterApi;
 import uk.gov.companieshouse.api.model.efs.submissions.SubmissionResponseApi;
 import uk.gov.companieshouse.efs.api.submissions.service.SubmissionService;
+import uk.gov.companieshouse.efs.api.submissions.model.PresenterApi;
```
Comment (hepsimo) [inline src/main/java/uk/gov/companieshouse/efs/api/submissions/controller/NewSubmissionController.java:18]: There's a possibility that Spring Boot bean validation isn't getting applied to the method
`newSubmission(@RequestBody @Valid @NotNull PresenterApi presenterApi, ...`
despite the `@Valid` annotation on parameter `presenterApi`

As an alternative I'd like to spike replacing these changes to this file with an `@Validated` annotation at the class level.

i.e.

```
@RestController
@Validated
@RequestMapping("/efs-submission-api/submissions/new")
public class NewSubmissionController {
```

Comment (mnycz-ch) [reply src/main/java/uk/gov/companieshouse/efs/api/submissions/controller/NewSubmissionController.java:18]: Hi @hepsimo - we can give it a try. I will make that change here.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by hepsimo — APPROVED** [review]
Approving this change to try out the approach of using @Validated

**Review by akunati-ch — APPROVED** [review]

**Pull request merged**